test/framework/integration_framework/integration_test_framework.cpp

```diff
@@ -222,6 +222,39 @@ namespace integration_framework {
     return *this;
   }
 
+  IntegrationTestFramework &IntegrationTestFramework::sendTx(
+      const shared_model::proto::Transaction &tx, std::set<TxStatus> statuses) {
+    std::condition_variable cv;
+    std::mutex m;
+
+    // statuses number that haven't being appeared yet
+    size_t count = statuses.size();
+    iroha_instance_->instance_->getStatusBus()
+        ->statuses()
+        .filter([&](auto s) { return s->transactionHash() == tx.hash(); })
+        .map([](auto s) { return TxStatus(s->get().which()); })
```

Comment:
dangerous cast from variant to enum. If someone changes the order of types in either variant or enum it will be hard to find the reason of the error. Maybe cast it explicitly via visit_in_place?